packages/sdk/src/evm/core/interfaces/contract.ts

```diff
@@ -6,4 +6,5 @@ import { NetworkOrSignerOrProvider } from "../types";
 export interface UpdateableNetwork {
   onNetworkUpdated(network: NetworkOrSignerOrProvider): void;
   getAddress(): string;
+  chainId: number;
```

Comment:
Oh, I see. Could have been getChainId() but whatevs

Reply:
yeah but since it's a getter only (u can never set it after constructing the class) a getter felt more correct (same with `getAddress()` to be fair) and `get getChainId()` seems excessive

Reply:
tbh should probably do the same with address, but not trying to change it in this same ark of work rn (also would have to keep it backwards compat anyways)